features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/SessionReplayConfiguration.kt

```diff
@@ -62,24 +60,13 @@ data class SessionReplayConfiguration internal constructor(
         }
 
         /**
-         * Sets the sample rate for Session Replay recorded Sessions. Please note that this
-         * sample rate will be applied on top of the already sampled in RUM session.
-         *
+         * Builds a [SessionReplayConfiguration] based on the current state of this Builder.
          * @param sampleRate must be a value between 0 and 100. A value of 0
          * means no session will be recorded, 100 means all sessions will be recorded.
          * The default value for the sample rate will be 0 meaning that no session replay will be
          * recorded if the sample rate will not be explicitly set.
          */
-        fun setSessionReplaySampleRate(@FloatRange(from = 0.0, to = 100.0) sampleRate: Float):
```

Comment:
Because this value is required when building the config there is no reason to have this method.

Reply:
Makes sense 👍. 

By removing this method we make the sample rate immutable, which is also the way we define it on iOS:
```swift
let sampleRate: Float // immutable & non-optional
```